### PR TITLE
W.I.P. build telescope base

### PR DESCRIPTION
**Pull request by StefWe**
Make build work with ubuntu 16.04
Cleanup the project
Add README

### CODE REVIEW

File: app/qml/telescope.qml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import QtQuick 2.4
+import Ubuntu.Components 1.3
+import Telescope 1.0
+
+/*!
+    \brief MainView with a Label and Button elements.
```
Comment (FlorentRevest) [inline app/qml/telescope.qml]: Remember to remove this comment from the template

**Comment by FlorentRevest** [comment]
Thanks for your efforts!
Ideally, the telescome repository should only contain the UT-specific QML UI.
The actual daemon keeping the synchronization alive and its corresponding library should be kept separate and maintained in https://github.com/AsteroidOS/asteroidsyncservice
I could find a lot of redundant code between the asteroidsyncservice repository and this PR. If you want to know how yo use asteroidsyncservice, you can have a look at the current starfish code.
This should give you an idea of the next step: https://github.com/AsteroidOS/starfish/commit/274d41299addc0ffb939fef064ac18e1a965b2dc

**Comment by StefWe** [comment]
Thank you so much for the review.
I'm trying to solve the points, but I'm a beginner that will need all some time.

> Ideally, the telescome repository should only contain the UT-specific QML UI.

I think asteroidsyncservice should be integrated as submodules. Then the daemon can be delivered with the .click package.

File: app/qml/telescope.qml (hunk 1 above)
Comment (StefWe) [reply app/qml/telescope.qml]: done

**Comment by FlorentRevest** [comment]
> Thank you so much for the review.
> I'm trying to solve the points, but I'm a beginner that will need all some time.

Np.

> > Ideally, the telescome repository should only contain the UT-specific QML UI.
> 
> I think asteroidsyncservice should be integrated as submodules. Then the daemon can be delivered with the .click package.

Agreed

**Comment by StefWe** [comment]
I try to build asteroidsyncservice as a submodule in telescope and have some trouble:
How can we trade this line?
https://github.com/AsteroidOS/asteroidsyncservice/blob/master/asteroidsyncserviced/asteroidsyncserviced.pro#L16
I didn't have this package in ubuntu touch.

Do we make this configurable or what are the possibilities here?
https://github.com/AsteroidOS/asteroidsyncservice/blob/master/asteroidsyncserviced/asteroidsyncserviced.pro#L21
https://github.com/AsteroidOS/asteroidsyncservice/blob/master/asteroidsyncserviced/main.cpp#L24
https://github.com/AsteroidOS/asteroidsyncservice/blob/master/asteroidsyncserviced/main.cpp#L31

**Comment by FlorentRevest** [comment]
For qt5-boostable, you can try just removing the line. The idea here is to have asteroidsyncserviced start from mapplauncherd which uses various "boosters" which are preloaded minimal Qt5 programs that make things faster. You don't need to use qt5-boostable, but you _may_ have to change a few things in main.cpp for it to work, namely, removing Q_DECL_EXPORT.

When it comes to the whole platform integration, remember that asteroidsyncserviced is a fork of rockpoold which, is based on rockworkd. There is some ubuntu platform integration code left there that I removed from asteroidsyncserviced in order not to have to maintain untouched code https://github.com/abranson/rockpool/tree/master/rockworkd/platformintegration/ubuntu this might serve as a good inspiration.

Ideally, there should be a qmake flag to select whether the platform integration should be Ubuntu or SailfishOS. This flag would affect the three links that you pasted.

**Comment by StefWe** [comment]
Thank you for your help!

I try to work on the https://github.com/AsteroidOS/asteroidsyncservice
There are submodules configured, but submodule init and update don't work. Is it just me?

**Comment by FlorentRevest** [comment]
I believe I wrote the file manually. Try to clone the repositories in .gitmodules manually.

**Comment by StefWe** [comment]
Yes that worked. Is it ok, if i made a pr to fix this or would you like to leave it like this?

**Comment by FlorentRevest** [comment]
Sure, please fix it.

**Pull request closed without merging**